pylsp/plugins/jedi_completion.py

```diff
@@ -62,36 +62,34 @@ def pylsp_completions(config, document, position):
         SNIPPET_RESOLVER.cached_modules = modules_to_cache_for
 
     include_params = snippet_support and should_include_params and use_snippets(document, position)
-    include_class_objects = snippet_support and should_include_class_objects and use_snippets(document, position)
-
-    ready_completions = [
-        _format_completion(
-            c,
-            include_params,
-            resolve=resolve_eagerly,
-            resolve_label_or_snippet=(i < max_to_resolve)
-        )
-        for i, c in enumerate(completions)
-    ]
-
-    # TODO split up once other improvements are merged
-    if include_class_objects:
-        for i, c in enumerate(completions):
-            if c.type == 'class':
-                completion_dict = _format_completion(
+    include_class_objects = (
+        snippet_support and should_include_class_objects and use_snippets(document, position)
+    )
+
+    ready_completions = []
+    for i, c in enumerate(completions):
+
+        ready_completions.append(
+            {
+                **_format_completion(
                     c,
-                    False,
+                    include_params,
                     resolve=resolve_eagerly,
-                    resolve_label_or_snippet=(i < max_to_resolve)
-                )
-                completion_dict['kind'] = lsp.CompletionItemKind.TypeParameter
-                completion_dict['label'] += ' object'
-                ready_completions.append(completion_dict)
+                    resolve_label_or_snippet=(i < max_to_resolve),
+                ),
+                'data': {'doc_uri': document.uri},
+            }
+        )
```

Comment:
Thank you for the attempt. I am afraid that it may not be faster in the general case.

Here the old code for the default settings was only using the list comprehension, which is generally faster [SO](https://stackoverflow.com/questions/30245397/why-is-a-list-comprehension-so-much-faster-than-appending-to-a-list) than doing append 1000 times. The advantage of merging the `include_class_objects` in there will be only felt by users who have this option enabled; I am not convinced that this option is even needed, but well it was kept for compatibility.

Also, please note that countless hours went into profiling this code for performance gains; I would be really more optimistic to see a proposed changeset derived from such similar profiling session rather than from intuition ;)